apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -260,12 +260,14 @@ export class ChangeAction extends BaseRushAction {
         existingChangeComments
       );
 
-      const email: string = this._changeEmailParameter.value
-        ? this._changeEmailParameter.value
-        : await this._detectOrAskForEmail(promptModule);
-      changeFileData.forEach((changeFile: IChangeFile) => {
-        changeFile.email = email;
-      });
+      if (this._isEmailRequired()) {
+        const email: string = this._changeEmailParameter.value
+          ? this._changeEmailParameter.value
+          : await this._detectOrAskForEmail(promptModule);
+        changeFileData.forEach((changeFile: IChangeFile) => {
+          changeFile.email = email;
```

Comment:
Although `includeEmailInChange` is specified per-project, this logic does not seem to consider that.

Here's how I tested it using the Rush Stack repo:

1. Modify `common/config/rush/version-policies.json` to add a second version policy with `includeEmailInChange=true`:
   ```js
   {
     "policyName": "rush",
     "definitionName": "lockStepVersion",
     "version": "5.47.0",
     "nextBump": "minor",
     "mainProject": "@microsoft/rush"
   },
   {
     "policyName": "other",
     "definitionName": "lockStepVersion",
     "version": "5.47.0",
     "nextBump": "minor",
     "includeEmailInChange": true
   }
   ```
2. Modify `rush.json` to assign this policy to a couple projects:
   ```js
    {
      "packageName": "@rushstack/module-minifier-plugin",
      "projectFolder": "webpack/module-minifier-plugin",
      "reviewCategory": "libraries",
      "versionPolicyName": "other"
    },
    {
      "packageName": "@rushstack/set-webpack-public-path-plugin",
      "projectFolder": "webpack/set-webpack-public-path-plugin",
      "reviewCategory": "libraries",
      "versionPolicyName": "other"
    }
   ```
3. Edit some .ts files to make changes under 3 different projects:
    - `apps\rush\src\start.ts` - a project using the "rush" version policy
    - `libraries\node-core-library\src\JsonFile.ts` - a project with no version policy, instead using `shouldPublish=true`
    - `webpack\module-minifier-plugin\src\MessagePortMinifier.ts` a project using the "other' version policy

4. Commit all those changes, then run `rush change`. (See [Contributing](https://rushjs.io/pages/contributing/) for instructions on how to invoke a local build of Rush.) It prompts for an email because `module-minifier-plugin` has `"includeEmailInChange": true`

**Expected:** Only the change file for `module-minifier-plugin` should store an email.

**Actual:** All 3 files store an email.

Reply:
Reproduced, I see it happening.